src/ws/session.c

```diff
@@ -362,6 +362,30 @@ map_gssapi_to_local (gss_name_t name,
   return str;
 }
 
+static bool
+acquire_service_credentials (gss_OID mech_type, gss_cred_usage_t usage, gss_cred_id_t *cred)
+{
+  /* custom credential store with our cockpit keytab */
+  static gss_key_value_element_desc store_elements[] = { { .key = "keytab", .value = COCKPIT_KTAB } };
+  static const gss_key_value_set_desc cockpit_ktab_store = { .count = 1, .elements = store_elements };
+  OM_uint32 major, minor;
+
+  debug ("acquiring cockpit service credentials");
+  major = gss_acquire_cred_from (&minor, GSS_C_NO_NAME, GSS_C_INDEFINITE, GSS_C_NO_OID_SET, usage,
+          (!getenv ("COCKPIT_TEST_KEEP_KTAB") && access (COCKPIT_KTAB, F_OK) == 0) ? &cockpit_ktab_store : NULL,
+          cred, NULL, NULL);
+
+  if (GSS_ERROR (major))
+    {
+      /* This is a routine error message, don't litter */
+      const char *msg = gssapi_strerror (mech_type, major, minor);
+      if (!strstr (msg, "nonexistent or empty"))
```

Comment:
I wonder if there's a better way to filter this error, considering gssapi has error codes. But it's just for debug messages so whatever.

Reply:
Sorry, you did this in the last commit. Marking as resolved.